frontend/src/components/Forms/AddPermitForm.js

```diff
@@ -75,87 +78,118 @@ const validateBusinessRules = (values) => {
   return errors;
 };
 
-export const AddPermitForm = (props) => (
-  <Form layout="vertical" onSubmit={props.handleSubmit}>
-    <Row gutter={16}>
-      <Col>
-        <Form.Item>
-          <Field
-            id="permit_type"
-            name="permit_type"
-            label="Permit type*"
-            placeholder="Select a permit type"
-            component={RenderSelect}
-            validate={[required]}
-            data={permitTypes}
-          />
-        </Form.Item>
-        {(props.permitTypeCode === "C" ||
-          props.permitTypeCode === "M" ||
-          props.permitActivityTypeCode === "X") && (
-          <Form.Item>
-            <Field
-              id="permit activity type*"
-              name="permit_activity_type"
-              label="Permit activity type*"
-              placeholder="Select a permit activity type"
-              component={RenderSelect}
-              data={permitActivityTypes}
-            />
-          </Form.Item>
-        )}
-        <Form.Item>
-          <Field
-            id="permit_no"
-            name="permit_no"
-            label="Permit number*"
-            component={RenderField}
-            validate={[required]}
-            inlineLabel={
-              props.permitTypeCode && `${props.permitTypeCode}${props.permitActivityTypeCode} -`
-            }
-          />
-        </Form.Item>
-        <Form.Item>
-          <Field
-            id="permit_status_code"
-            name="permit_status_code"
-            label="Permit status*"
-            placeholder="Select a permit status"
-            component={RenderSelect}
-            data={props.permitStatusOptions}
-            validate={[required]}
-          />
-        </Form.Item>
-        <Form.Item>
-          <Field
-            id="issue_date"
-            name="issue_date"
-            label="Issue date*"
-            component={RenderDate}
-            validate={[required, dateNotInFuture]}
-          />
-        </Form.Item>
-      </Col>
-    </Row>
-    <div className="right center-mobile">
-      <Popconfirm
-        placement="topRight"
-        title="Are you sure you want to cancel?"
-        onConfirm={props.closeModal}
-        okText="Yes"
-        cancelText="No"
-      >
-        <Button className="full-mobile" type="secondary">
-          Cancel
-        </Button>
-      </Popconfirm>
-      <Button className="full-mobile" type="primary" htmlType="submit" disabled={props.submitting}>
-        {props.title}
-      </Button>
-    </div>
-  </Form>
-);
+class AddPermitForm extends Component {
+  onFileLoad = (fileName, document_manager_guid) => {
+    this.props.initialValues.uploadedFiles.push({ fileName, document_manager_guid });
```

Comment:
Yup, it's intended to used as you have described, what we have done is piggybacked the files into the form-data (through initialValues). The handleSubmit accepts one parameter which is the data on the form, so without doing this, I don't think we can access the list of files. 

You have sparked me to do some investigating and it looks like the handleSubmit handlers have other parameters that can be passed to them. Which would allow us to keep them separate (props of the form?) and address your concerns. I'll edit when i find something.

Reply:
This stood out to me as undesired as well, although the last time I looked into redux forms there didn't seem to be a way of handling hidden fields other than some hacky workarounds.

@Jsyro if there is now a better way to handle this, let's change this.

Reply:
Changed it to use change actions and local state, take a look and let me know your thoughts.

Reply:
Changes in `frontend/src/components/Forms/PermitAmendmentForm.js` look pretty good to me now. Are you able to update this file to work the same way?

Reply:
Woops, sorry about that, applied it to this form as well.